js/src/components/Tournament.vue

```diff
@@ -70,94 +72,82 @@ export default {
-          var runnerupName = t.runnerups[i]
-          var player = t.players[j]
-
-          if (runnerupName === player.name) {
-            ret.push(player)
-          }
-        }
+        return []
       }
 
-      return ret
+      return _.map(t.runnerups, (runnerupName) => {
+        return _.find(t.players, { name: runnerupName })
+      })
     }
   },
 
   methods: {
     start: function () {
-      this.$http.get('/api/towerfall/' + this.$data.tournament.id + '/start/').then((res) => {
-        console.log(res)
-        var j = res.json()
-        this.$route.router.go('/towerfall' + j.redirect)
-        // XXX: Worst hack of all time
-        this.$data.tournament.started = 'hehe'
-      }, (res) => {
-        console.log('fail')
-        console.log(res)
-      })
+      if (this.tournament) {
+        this.api.start({ id: this.tournament.id }).then((res) => {
+          console.log("start response:", res)
+          let j = res.json()
+          this.$route.router.go('/towerfall' + j.redirect)
+        }, (err) => {
+          console.error(`start for ${this.tournament} failed`, err)
+        })
+      } else {
+        console.error("start called with no tournament")
+      }
     },
     next: function () {
-      this.$http.get('/api/towerfall/' + this.$data.tournament.id + '/next/').then((res) => {
-        console.log(res)
-        var j = res.json()
-        this.$route.router.go('/towerfall' + j.redirect)
-      }, (res) => {
-        console.log('fail')
-        console.log(res)
-      })
+      if (this.tournament) {
+        this.api.next({ id: this.tournament.id }).then((res) => {
+          console.debug("next response:", res)
+          let j = res.json()
+          this.$route.router.go('/towerfall' + j.redirect)
+        }, (err) => {
+          console.error(`next for ${this.tournament} failed`, err)
+        })
+      } else {
+        console.error("next called with no tournament")
+      }
+    }
+  },
+
+  created: function () {
+    console.debug("Creating API resource")
+    let customActions = {
+      start: { method: "GET", url: "/api/towerfall{/id}/start/" },
+      next: { method: "GET", url: "/api/towerfall{/id}/next/" },
+      getData: { method: "GET", url: "/api/towerfall/tournament{/id}/" }
     }
+    this.api = this.$resource("/api/towerfall", {}, customActions)
   },
 
   route: {
     data ({ to }) {
-      // We need a reference here because `this` inside the callback will be
-      // the main App and not this one.
-      var $vue = this
-
-      to.router.app.$watch('tournaments', function (newVal, oldVal) {
-        for (var i = 0; i < newVal.length; i++) {
-          if (newVal[i].id === to.params.tournament) {
-            console.log("watch update")
-            console.log(newVal[i])
-            $vue.$set('tournament', newVal[i])
-          }
-        }
+      // listen for tournaments from App
+      this.$on(`tournament${to.params.tournament}`, (tournament) => {
```

Comment:
It's the ES6 [string interpolation](https://developer.mozilla.org/en/docs/Web/JavaScript/Reference/Template_literals) syntax. Basically, you surround your string with backticks, which allows you to evaluate expressions within ${}.